server/server.go

```diff
@@ -66,16 +66,21 @@ func NewServer(host string, port int, router *store.Router, log log.Logger,
 func WithMetrics(handleFn func(http.ResponseWriter, *http.Request) error,
 	m metrics.Metricer) func(http.ResponseWriter, *http.Request) error {
 	return func(w http.ResponseWriter, r *http.Request) error {
-		// we use a commitment schema (https://github.com/Layr-Labs/eigenda-proxy?tab=readme-ov-file#commitment-schemas)
-		// where the first 3 bytes of the path are the commitment header
-		// commit type | da layer type | version byte
-		// we want to group all requests by commitment header, otherwise the prometheus metric labels will explode
-		// TODO: commitment header is different for non-op commitments. We will need to change this to accommodate other commitments.
-		//       probably want (commitment mode, cert version) as the labels, since commit-type/da-layer are not relevant anyways.
-		commitmentHeader := r.URL.Path[:3]
-		recordDur := m.RecordRPCServerRequest(commitmentHeader)
+		// label requests with commitment mode and version
+		ct, err := ReadCommitmentMode(r)
+		if err != nil {
+			m.RecordBadRequestHeader(r.Method, "invalid commitment mode")
+			return err
+		}
+		vb, err := ReadCommitmentVersion(r, ct)
+		if err != nil {
+			m.RecordBadRequestHeader(r.Method, "invalid commitment version")
+			return err
+		}
```

Comment:
nice - looks like this operation is done twice now though. Maybe we could extend the `handleFunc` to take these fields in to avoid redundant operations.